T15723

tchx84 opened the pull request:
https://phabricator.endlessm.com/T15723

ptomato commented on ekncontent/tests/ekncontent/testAudioObjectModel.js:
```diff
@@ -0,0 +1,44 @@
+const InstanceOfMatcher = imports.tests.InstanceOfMatcher;
+
+const MOCK_AUDIO_DATA = {
+    '@id': 'ekn://BagRaiders/ShootingStars',
```
Better to use a valid EKN ID here in case later we add validation somewhere in the pipeline.

ptomato commented on js/app/modules/card/audio.js:
```diff
@@ -0,0 +1,62 @@
+        this.set_label_or_hide(this._synopsis_label, this.model.synopsis);
+
+        this._title_label.visible = this.show_title;
+        let player = new EosKnowledgePrivate.MediaBin({ audioMode: true });
```
For consistency, use the underscore format

ptomato commented on js/app/modules/card/audio.js:
```diff
@@ -0,0 +1,62 @@
+        player.get_style_context().add_class(Utils.get_element_style_class(Audio, 'player'));
+        player.set_uri(this.model.ekn_id)
+        player.show_all();
+        this.attach(player, 1, 1, 1, 1);
```
This seems the wrong place to attach it.

ptomato commented on js/app/modules/card/audio.js:
```diff
@@ -0,0 +1,62 @@
+
+    Properties: {
+        /**
+         * Property: show-titles
```
Doesn't match the property name

ptomato commented on js/app/modules/card/audio.js:
```diff
@@ -0,0 +1,62 @@
+const Utils = imports.app.utils;
+
+/**
+ * Class: Audio
```
Going forward, we really need better documentation at the time we introduce new modules, because one thing that will never happen is going back and making a comprehensive review of documentation. Could you please make sure that this comment and the property below make sense as part of the modular framework API documentation?

tchx84 commented:
I have addressed these comments and split this PR into https://github.com/endlessm/eos-knowledge-lib/pull/1392 and https://github.com/endlessm/eos-knowledge-lib/pull/1393.

Closed.
